src/types/mutex.rs

```diff
@@ -0,0 +1,86 @@
+use std::ops::Deref;
+
+/// Wrapper for [`PyMutex`](https://docs.python.org/3/c-api/init.html#c.PyMutex) exposing an RAII interface.
+#[derive(Debug)]
+pub struct PyMutex<T> {
+    _mutex: crate::ffi::PyMutex,
+    data: T,
+}
```

Comment:
```suggestion
pub struct PyMutex<T> {
    _mutex: UnsafeCell<crate::ffi::PyMutex>,
    data: UnsafeCell<T>,
}
```

- both cells are needed; one for lock / unlock, and one for data to be readable from interior mutability.